psh/test-history.py

```diff
@@ -0,0 +1,106 @@
+from psh.tools.common import CHARS, get_rand_names
+import psh.tools.psh as psh
+import string
+
+
+def assert_sent(p, cmds, msg='', multi=False):
+    i = 2
+    expected = ['   1  history']
+    for cmd in cmds:
+        for _ in range(3 if multi else 1):
+            p.sendline(cmd)
+            psh.assert_prompt(p, msg=f'No prompt seen after passing the following command: {cmd}')
+        if i >= 10:
+            line = f'  {i}  ' + cmd
+        else:
+            line = f'   {i}  ' + cmd
+        expected.append(line)
+        i = i + 1
+    expected.append(f'  {i}  history')
+    psh.assert_cmd(p, 'history', expected, msg)
+
+
+def assert_help(p):
+    assert_msg = "Prompt hasn't been displayed properly, when calling with -h argument"
+    psh.assert_prompt_after_cmd(p, 'history -h', assert_msg)
+
+    assert_msg = "Prompt hasn't been displayed properly, when calling with wrong arguments"
+    psh.assert_prompt_after_cmd(p, 'history -z', assert_msg)
```

Comment:
I checked and `history -z` prints usage output, but `history --z` prints nothing. Do you know why?

Reply:
It's the matter of current implementation. getopt() returns `-1`, when `--` is pased and optind is incremented, so nothing happens:
https://github.com/phoenix-rtos/phoenix-rtos-utils/blob/94a09ff0f89b2feda06ac9a9719ea4c949fc71e3/psh/pshapp/pshapp.c#L930-L952